lab_6_pipeline/pipeline.py

```diff
@@ -18,21 +37,55 @@ def __init__(self, path_to_raw_txt_data: Path):
         """
         Initializes CorpusManager
         """
+        self._storage = {}
+        self.path = path_to_raw_txt_data
+        self._validate_dataset()
+        self._scan_dataset()
 
     def _validate_dataset(self) -> None:
         """
         Validates folder with assets
         """
+        if not isinstance(self.path, Path) or not self.path.exists():
+            raise FileNotFoundError
+
+        if not self.path.is_dir():
+            raise NotADirectoryError
+
+        if not self.path.glob("*"):
+            raise EmptyDirectoryError
+
+        meta_list = [elem.name for elem in self.path.glob("*_meta.json")]
+        raw_list = [elem.name for elem in self.path.glob("*_raw.txt")]
+
+        meta_ids = sorted([int(re.search(r'\d+', elem)[0]) for elem in meta_list])
```

Comment:
what if there are text files with non-numeric elements instead of id?

Reply:
done